vita3k/shader/src/translator/alu.cpp

```diff
@@ -1357,8 +1357,12 @@ bool USSETranslatorVisitor::vdual(
         return false;
     }
 
-    Imm4 op1_write_mask = prim_ustore ? get_dual_op_write_mask(op1_info, op1.opr.dest.bank == RegisterBank::FPINTERNAL) : write_mask_non_gpi;
-    Imm4 op2_write_mask = prim_ustore ? write_mask_non_gpi : get_dual_op_write_mask(op2_info, op2.opr.dest.bank == RegisterBank::FPINTERNAL);
+    Imm4 fixed_write_mask = write_mask_non_gpi | ((comp_count_type && dual_op1_ext_vec3_or_has_w_vec4) << 3u);
```

Comment:
For reference, dual_op1_ext_vec3_or_has_w_vec4 is a bit that tells us if the w component is available for operating on DUAL4